lib/redis/event_handler.js

```diff
@@ -39,6 +39,25 @@ exports.connectHandler = function (self) {
 
     self.initParser();
 
+    self.lastWriteTime = null;
+    if (typeof self.options.timeoutPerCommand === 'number') {
+      var timeoutPerCommand = self.options.timeoutPerCommand;
+      debug('Per-command timeout set to %s', self.options.timeoutPerCommand);
+      self.timeoutPerCommand = setInterval(() => {
+        var commandQueueLength = self.commandQueue.length;
+        if (
+          commandQueueLength > 0 &&
+          Date.now() - self.lastWriteTime > self.options.timeoutPerCommand
+        ) {
+          debug('Command timed out. Pending commands: %s', commandQueueLength);
+          var err = new Error('Command timed out');
+          self.flushQueue(err, { offlineQueue: false });
+          self.silentEmit('error', err);
+          self.disconnect(true);
```

Comment:
I assume that this will trigger an automatic reconnect to that server, correct?

Reply:
Good, but see my follow-up to #587.  If this node was part of a cluster, `retryStrategy` is overridden and set to `null` so there would in fact be no automatic reconnect to that node.

Reply:
Per #587, there will not be any automatic reconnect in a cluster environment.